mlcomp/parallelm/ml_engine/sagemaker_engine.py

```diff
@@ -0,0 +1,111 @@
+        # which setup the shared credentials and region configuration
+        global boto3, ClientError
+        import boto3
+        from botocore.exceptions import ClientError
+
+        self._setup_logger(eng_args_config)
+        self._setup_iam_role(eng_args_config)
+
+    @property
+    def iam_role(self):
+        return self._iam_role
+
+    def _setup_iam_role(self, eng_args_config):
+        self._iam_role = EeArg(eng_args_config.get('iam_role')).value
+        if not self._iam_role:
+            # acount_id = boto3.client('sts').get_caller_identity().get('Account')
+
+            now = datetime.datetime.now()
+            role_name = 'AmazonSageMaker-ExecutionRole-{y}{month}{d}T{h}{minute}{s}' \
+                .format(y=now.year, month=now.month, d=now.day, h=now.hour, minute=now.minute, s=now.second)
+
+            tags = [{'Key': self._tag_key, 'Value': self._tag_value}] if self._tag_key else []
+
+            client = boto3.client('iam')
+
+            try:
+                response = client.create_role(
+                    Path='/service-role/',
+                    RoleName=role_name,
+                    AssumeRolePolicyDocument='<URL-encoded-JSON>',
+                    Description='Auto generated sagemaker aim role by ParallelM',
+                    Tags=tags
+                )
+                self._iam_role_name = response['Role']['RoleName']
+                self._iam_role = response['Role']['Arn']
+            except ClientError as e:
+                self._logger.error("Failed to create a an iam role for sagemaker service!\n{}".format(e))
+                raise e
+
+    def _read_execution_env_params(self):
+        ee_config = self._pipeline.get('executionEnvironment', dict()).get('configs')
+        if not ee_config:
+            raise MLCompException("Missing execution environment section in pipeline json!")
+
+        eng_config = ee_config.get('engConfig')
+        if not eng_config:
+            raise MLCompException("Missing execution environment engine section in pipeline json!")
+
+        if eng_config['type'] != 'sagemaker':
+            raise MLCompException("Unexpected engine type in execution environment! expected: 'sagemake', got: {}"
+                                  .format(eng_config['type']))
+
+        return eng_config['arguments']
+
+    def _setup_env(self, eng_args_config):
+        region = EeArg(eng_args_config.get('region')).value
+
+        aws_access_key_id = EeArg(eng_args_config.get('aws_access_key_id')).value
+        if not aws_access_key_id:
+            raise MLCompException("Empty 'aws_access_key_id' parameter in execution environment!")
+
+        aws_secret_access_key = EeArg(eng_args_config.get('aws_secret_access_key')).value
+        if not aws_secret_access_key:
+            raise MLCompException("Missing 'aws_secret_access_key' parameter in execution environment!")
+
+        os.environ['AWS_DEFAULT_REGION'] = region  # 'us-west-2'
```

Comment:
Define keys as constants in the beginning of the file.

Reply:
Fixed